lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -3318,6 +3318,46 @@ Expr *TypeChecker::coerceToRValue(Expr *expr,
   return expr;
 }
 
+bool TypeChecker::convertToType(Expr *&expr, Type type, constraints::ConstraintSystem& cs,
```

Comment:
I don't think this needed to be extracted.  The [coerceToType](https://github.com/apple/swift/pull/24511/files#diff-e38c4e32f8ef5d6750d4c6a27b6b5397R2871) change you made above is the answer we were discussing earlier.  There's one more caller in CSApply that should also be changed

```diff
-if (tc.convertToType(sub, toType, cs.DC))
-  return nullptr;
+sub = coerceToType(sub, toType, cs.getConstraintLocator(sub));
```

Reply:
That makes sense, I thought `coerceToType` was only available on the solution, so I extracted because of that, to solve it and have the solution available. But here at this point, it's already salvaging a system that it couldn't solve, so use this here and call solve() will be to re-typecheck right? 
Sorry for all these questions I'm just trying to check if I'm understanding it all correctly
Btw, @CodaFi  thank's for all the comments :)) 
Making the changes then